### PR TITLE
feat: evaluate the modulo operator in linker scripts

### PR DESCRIPTION
**Pull request by vishruth-thimmaiah**
Implement the modulo operator(`%`) in arithmetic operations in linker scripts.

### CODE REVIEW

File: wild/tests/sources/elf/linker-script-assert-pass/linker-script-assert-pass.ld [hunk 1]
```diff
@@ -14,6 +14,7 @@ ASSERT(1 == 1, "1 == 1 must always be true");
 
 /* Arithmetic assertion */
 ASSERT(1 + 1 == 2, "arithmetic must work");
+ASSERT(12 % 10 == 2, "modulo must work");
```
Comment (davidlattimore) [inline wild/tests/sources/elf/linker-script-assert-pass/linker-script-assert-pass.ld]: Can you also test that we have the same behaviour as the other linkers with regard to negative numbers?

Comment (vishruth-thimmaiah) [reply wild/tests/sources/elf/linker-script-assert-pass/linker-script-assert-pass.ld]: Fixed, I've modified the PR to use `wrapping_rem` instead, I also fixed division with negative numbers.

**Review by davidlattimore — APPROVED** [review]
